app/controllers/api/v0/events_controller.rb

```diff
@@ -6,6 +6,13 @@ def create
     render(json: error, status: error.status_code) and return if error
 
     inbound_binding.events.each do |event|
+      # first, schema validate the event.data object
+
+      # second, inject the user_uuid if present
+      if current_user_uuid && !event.data.key?(:user_uuid)
+        event.data[:user_uuid] = current_user_uuid
+      end
```

Comment:
But now I see the issue: we're _only_ storing in the topic the schema-matching data as sent into the API, so it would not be available when reading from the topic. Hmm.

How about we go ahead and make having/allowing user_uuid a schema requirement, and if the upstream sends data with a non-empty value for it, it must match what is extracted from the cookie, otherwise, we error.

Reply:
`AsyncKafkaProducer.instance.produce(data, topic: topic)` is the interface.  there's no way (afaik) to pass in metadata to the request.  my thought here is that user_uuid would be optional, or a schema requirement.  we certainly can remove the override part of this..i wasnt sure about that, but your review of it makes sense (to not allow override).  we actually do have another override..for loadtesting.  can remove this check.

Reply:
cool, though I think we should probably test to see if the user is _trying_ to pass in a bad user_uuid and error. Or do you think we should just stomp on it?

Reply:
not sure.  e.g., thru loadtesting, we'll want it to get a particular user_uuid and not worry about cookies, which makes load testing harder.  but we have a way for that.   so..perhaps we can just stomp it.

Reply:
"How about we go ahead and make having/allowing user_uuid a schema requirement".  +1 <-